services/QuillLMS/app/models/teacher_notification_setting.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+# == Schema Information
+#
+# Table name: teacher_notification_settings
+#
+#  id                :bigint           not null, primary key
+#  notification_type :text             not null
+#  created_at        :datetime         not null
+#  updated_at        :datetime         not null
+#  user_id           :bigint           not null
+#
+# Indexes
+#
+#  index_teacher_notification_settings_on_user_id           (user_id)
+#  index_teacher_notification_settings_on_user_id_and_type  (user_id,notification_type) UNIQUE
+#
+# Foreign Keys
+#
+#  fk_rails_...  (user_id => users.id)
+#
+class TeacherNotificationSetting < ApplicationRecord
+  DEFAULT_FOR_NEW_USERS = [
+    TeacherNotification::StudentCompletedDiagnostic,
+    TeacherNotification::StudentCompletedAllDiagnosticRecommendations,
+    TeacherNotification::StudentCompletedAllAssignedActivities
+  ].map(&:name).freeze
+
+  def self.notification_types
+    TeacherNotification.subclasses.map(&:name)
```

Comment:
Because of how Rails lazy loads classes, I don't think this will [always](https://guides.rubyonrails.org/autoloading_and_reloading_constants.html#single-table-inheritance) work.

As a stopgap until we update to zeitwerk, I've been [hard-coding](https://github.com/empirical-org/Empirical-Core/blob/develop/services/QuillLMS/app/models/provider_classroom_user.rb#L23) the subclass names into the base class